packages/headless/src/modules/brands/repositories/index.ts

```diff
@@ -0,0 +1,35 @@
+import { Product } from '../../../entity/product';
+import { IBrandDatabase } from './brand.database.interface';
+import { Injectable } from '@nestjs/common';
+import { Brand } from 'src/entity/brand';
+
+@Injectable()
+export class BrandRepository {
+    constructor(private readonly db: IBrandDatabase) { }
+
+    async findBrandById(brandId: string): Promise<Brand | null> {
+        const brand = await this.db.getBrandById(brandId);
```

Comment:
don't use mongoose default _id

Reply:
if we are not searching with mongoose _id then what are we searching it with, as no other database system is still injected